services/blockstorage/adapter/filesystem/file_system_persistence.go

```diff
@@ -242,6 +245,9 @@ func buildIndex(r io.Reader, firstBlockOffset int64, logger log.Logger, c blockC
 		if err != nil {
 			return nil, errors.Wrap(err, "failed building block height index")
 		}
+		if metrics != nil {
```

Comment:
no. remove the if , in all the tests use NewRegistry()
not a good idea that in production code you allow a value to be null and ignore just to make your tests easier.